NOISSUE - Add Identify gRPC method

anovakovic01 opened the pull request:
Add `Identify` gRPC method that is needed for MQTT. Add tests for `Identify`.

Commit:
Add identify gRPC method to things service

Signed-off-by: Aleksandar Novakovic <aleksandar.novakovic@mainflux.com>

Commit:
Add test for identify endpoint

Signed-off-by: Aleksandar Novakovic <aleksandar.novakovic@mainflux.com>

mijicd commented on things/api/grpc/client.go:
```diff
@@ -12,20 +12,30 @@ var _ mainflux.ThingsServiceClient = (*grpcClient)(nil)
 	).Endpoint()
 
-	return &grpcClient{endpoint}
+	ie := kitgrpc.NewClient(
```
Let's inline both of these:

```golang
return &grpcClient{
    canAccess: ...,
    identify: ...
}
```

mijicd commented on things/api/grpc/endpoint.go:
```diff
@@ -15,8 +15,19 @@ func canAccessEndpoint(svc things.Service) endpoint.Endpoint {
+		req := request.(identifyReq)
+		id, err := svc.Identify(req.thingKey)
+		if err != nil {
+			return identityRes{id: "", err: err}, err
```
Please remind me why do we need to keep both "copies" (in structure, and as return value)?

mijicd commented on things/api/grpc/endpoint_test.go:
```diff
@@ -79,3 +79,33 @@ func TestCanAccess(t *testing.T) {
+	svc := newService(map[string]string{token: email})
+	startGRPCServer(svc, port)
+
+	th, _ := svc.AddThing(token, thing)
```
As far as I recall, `sth` is used in remaining tests.

mijicd commented on things/api/grpc/requests.go:
```diff
@@ -16,3 +16,7 @@ func (req accessReq) validate() error {
 }
+
+type identifyReq struct {
+	thingKey string
```
`key` is sufficient.

mijicd commented on things/api/grpc/server.go:
```diff
@@ -12,24 +12,40 @@ import (
 // NewServer returns new ThingsServiceServer instance.
 func NewServer(svc things.Service) mainflux.ThingsServiceServer {
-	handler := kitgrpc.NewServer(
+	cah := kitgrpc.NewServer(
```
See the remark given above for client.

mijicd commented on things/api/grpc/server.go:
```diff
@@ -12,24 +12,40 @@ import (
 
 func (s *grpcServer) CanAccess(ctx context.Context, req *mainflux.AccessReq) (*mainflux.Identity, error) {
-	_, res, err := s.handler.ServeGRPC(ctx, req)
+	_, res, err := s.canAccess.ServeGRPC(ctx, req)
```
Let's rename to either `gs` or `svc`.

mijicd commented on things/postgres/things.go:
```diff
@@ -98,3 +98,16 @@ func (tr thingRepository) Remove(owner, id string) error {
 	return nil
 }
+
+func (tr thingRepository) Identify(key string) (string, error) {
```
Instead of naming it `Identify`, I'd rather propose that we rename all repository methods. However, that would make this PR quite big, therefore I propose that you rename only this method to: `RetrieveByKey`.

If you decide to implement the "full rename", replace `All` with `RetrieveAll()` and `One` with `RetrieveByID`.

Commit:
Fix http and ws tests

Update things service mocks in http and ws services.

Signed-off-by: Aleksandar Novakovic <aleksandar.novakovic@mainflux.com>

Commit:
Rename repository methods in accordance with changes

Rename One to RetrieveByID. Rename All to RetrieveAll. Rename
Identify to RetrieveByKey.

Signed-off-by: Aleksandar Novakovic <aleksandar.novakovic@mainflux.com>

mijicd commented on things/things.go:
```diff
@@ -37,17 +37,17 @@ type ThingRepository interface {
 
-	// Identify returns thing ID for given thing key.
-	Identify(string) (string, error)
+	// RetrieveByKey returns thing ID for given thing key.
```
Move it close to the other retrievals.

mijicd commented on things/api/grpc/client.go:
```diff
@@ -17,25 +17,24 @@ type grpcClient struct {
+	return &grpcClient{
+		canAccess: kitgrpc.NewClient(
+			conn,
+			"mainflux.ThingsService",
```
Please extract to function-local constant.

Commit:
Move RetrieveByKey method close to other retrieve methods

Signed-off-by: Aleksandar Novakovic <aleksandar.novakovic@mainflux.com>

Commit:
Rename retrieval methods in channel repository

Signed-off-by: Aleksandar Novakovic <aleksandar.novakovic@mainflux.com>

Commit:
Rename retrieval method in user repository

Signed-off-by: Aleksandar Novakovic <aleksandar.novakovic@mainflux.com>
